### PR TITLE
⚠ remove olm.bundle.mediatype property and clusterextension support for plain+v0

### PR DESCRIPTION
**Pull request by joelanford**
Fixes #822

# Description

## Reviewer Checklist

- [ ] API Go Documentation
- [ ] Tests: Unit Tests (and E2E Tests, if appropriate)
- [ ] Comprehensive Commit Messages
- [ ] Links to related GitHub Issue(s)

### CODE REVIEW

**Review by everettraven — CHANGES_REQUESTED** [review]
Should probably remove the plain-v0 entries under testdata/bundles as well

(Apologies if I missed the removal - on GH mobile so it's entirely possible I missed it being removed)

File: internal/controllers/clusterextension_controller.go [hunk 1]
```diff
@@ -135,25 +135,16 @@ func (r *ClusterExtensionReconciler) reconcile(ctx context.Context, ext *ocv1alp
...
 
-	bundleProvisioner, err := mapBundleMediaTypeToBundleProvisioner(mediaType)
-	if err != nil {
-		setInstalledStatusConditionFailed(&ext.Status.Conditions, err.Error(), ext.GetGeneration())
-		setDeprecationStatusesUnknown(&ext.Status.Conditions, "deprecation checks have not been attempted as installation has failed", ext.GetGeneration())
-		return ctrl.Result{}, err
-	}
+	// Assume all bundles are registry+v1 for now, since that's all we support.
+	// If the bundle is not a registry+v1 bundle, the conversion will fail.
+	bundleProvisioner := "core-rukpak-io-registry"
```
Comment (m1kola) [inline internal/controllers/clusterextension_controller.go:146]: Nit: Maybe let's get rid of `bundleProvisioner` variable and of param in `r.GenerateExpectedBundleDeployment`?

Will help to reduce the noise in `reconcile` a little bit.

**Review by m1kola — COMMENTED** [review]
What do we do with [this demo](https://github.com/operator-framework/operator-controller/blob/5985cce3b9dd8d9d0f53b926c242202d9a3dbccb/docs/demos/coast-to-coast-q4-2023.md)? It seem to rely on `plain+v0` bundle types. I think this PR will break the demo.

**Review by everettraven — APPROVED** [review]

**Comment by everettraven** [comment]
>What do we do with [this demo](https://github.com/operator-framework/operator-controller/blob/5985cce3b9dd8d9d0f53b926c242202d9a3dbccb/docs/demos/coast-to-coast-q4-2023.md)? It seem to rely on plain+v0 bundle types. I think this PR will break the demo.

I'd vote to either:
- Remove it entirely since we have shifted direction quite significantly from when that was written
- Update it accordingly

I personally am leaning towards removal since we are going to need to write new coast-to-coast documentation/tutorials for the new state of the world

**Comment by joelanford** [comment]
> I personally am leaning towards removal

I'll remove it. I was having similar thoughts as well.

**Review by m1kola — APPROVED** [review]

**Pull request merged**